Chat_App/Core/CoreData/CoreDataManager.swift

```
@@ -0,0 +1,53 @@
+//
+//  ChatModel.swift
+//  test
+//
+//  Created by Ana Dzamelashvili on 5/17/23.
+//
+
+import Foundation
+import UIKit
+import CoreData
+
+class CoreDataManager {
+    
+    let context = (UIApplication.shared.delegate as! AppDelegate).persistentContainer.viewContext
+    let request: NSFetchRequest<MessageEntity> = MessageEntity.fetchRequest()
+    
+    //    MARK: - Load messages from coreData
+    func loadObjects<T: NSManagedObject>(with request: NSFetchRequest<T>) -> [T]? {
```

better would be making such functions as `throws` and create error system for CoreDataManager with enums and localized descriptions (or at least just descriptions) and throw those errors from these functions instead of `print` and `return nil`. Cause that way you'll be able to use it via `do-try-catch` and in `catch` block on the side you're calling these functions you'll be able to present corresponding error message on a screen (such as like "Sorry! We weren't be able to load your messages history!")